texar/torch/modules/pretrained/pretrained_bert.py

```diff
@@ -70,59 +83,86 @@ def _transform_config(cls, cache_dir: str) -> Dict[str, Any]:
         for file in files:
             if file.endswith('config.json'):
                 config_path = os.path.join(root, file)
+                with open(config_path) as f:
+                    config_ckpt = json.loads(f.read())
+                    hidden_dim = config_ckpt['hidden_size']
+                    vocab_size = config_ckpt['vocab_size']
+                    type_vocab_size = config_ckpt['type_vocab_size']
+                    position_size = config_ckpt['max_position_embeddings']
+                    embedding_dropout = config_ckpt['hidden_dropout_prob']
+                    num_blocks = config_ckpt['num_hidden_layers']
+                    num_heads = config_ckpt['num_attention_heads']
+                    dropout_rate = config_ckpt['attention_probs_dropout_prob']
+                    residual_dropout = config_ckpt['hidden_dropout_prob']
+                    intermediate_size = config_ckpt['intermediate_size']
+                    hidden_act = config_ckpt['hidden_act']
+
+            elif file.endswith('model.pt'):
```

Comment:
This check looks a bit hacky. I think it might be better to refactor the interfaces and also pass in `pretrained_model_name` to `_transform_config` and `_init_from_checkpoint`.